File: ontrack-extension-jenkins/src/main/java/net/nemerosa/ontrack/extension/jenkins/JenkinsExtensionFeature.java
package net.nemerosa.ontrack.extension.jenkins;

import net.nemerosa.ontrack.extension.support.AbstractExtensionFeature;
import net.nemerosa.ontrack.model.extension.ExtensionFeatureOptions;
import org.springframework.stereotype.Component;

@Component
public class JenkinsExtensionFeature extends AbstractExtensionFeature {

    public JenkinsExtensionFeature() {
        super("jenkins", "Jenkins", "Provides links with Jenkins", ExtensionFeatureOptions.DEFAULT.withGui(true));
    }

}

File: ontrack-extension-jenkins/src/main/java/net/nemerosa/ontrack/extension/jenkins/JenkinsExtensionFeature.kt
package net.nemerosa.ontrack.extension.jenkins

import net.nemerosa.ontrack.extension.indicators.IndicatorsExtensionFeature
import net.nemerosa.ontrack.extension.scm.SCMExtensionFeature
import net.nemerosa.ontrack.extension.support.AbstractExtensionFeature
import net.nemerosa.ontrack.model.extension.ExtensionFeatureOptions
import org.springframework.stereotype.Component

@Component
class JenkinsExtensionFeature(
    indicatorsExtensionFeature: IndicatorsExtensionFeature,
    scmExtensionFeature: SCMExtensionFeature,
) : AbstractExtensionFeature(
    id = "jenkins",
    name = "Jenkins",
    description = "Provides links with Jenkins",
    options = ExtensionFeatureOptions.DEFAULT
        .withGui(true)
        .withDependency(scmExtensionFeature)
        .withDependency(indicatorsExtensionFeature)
)